ui/src/app/pages/homepage/homepage.tsx

```diff
@@ -156,16 +176,38 @@ export const Homepage = withUserProfile()(class extends React.Component<Props, S
     }
   }
 
+  async validateRasLoginGovLink() {
+    const code = (new URL(window.location.href)).searchParams.get('code');
+    const redirectUrl = buildRasRedirectUrl();
+    if (code) {
+      this.setState({rasLoginGovLoading: true});
```

Comment:
Where doe sthis get set to `false` in the happy path?

Reply:
I my understanding is the card will has
```
    completionPropsKey: 'rasLoginGovLinked',
    loadingPropsKey: 'rasLoginGovLoading',
```
And if `completionPropsKey` is true, that button will become `rasLoginGovLinked` and grey out.
(at least that is what eRA Commons has).

What I added here is setting this to false in unhappy case. In eRA scenario, if link failed, user may have to refresh the page to be able to click the button again. 
I am not sure if I understand this correctly, I never tried eRA, but at lease to RAS, without this, I have to refresh the page. Do you think we should/can update eRA one as well?

Reply:
Well, this spinner should be getting shown after you land back on the dashboard from RAS, but based on my rereading of the code, I guess as long as the task is "complete", the loading status is basically ignored. I guess this is fine for consistency.